include/flatbuffers/flexbuffers.h

```diff
@@ -688,6 +691,31 @@ class Reference {
   Type type_;
 };
 
+// Template specialization for As().
+template<> inline bool Reference::As<bool>() { return AsBool(); }
```

Comment:
Why are these not inside `Reference` ?

Reply:
AFAIK, template specialization can't be added inside class scope :/

http://rextester.com/PDUXKQ82485